### PR TITLE
Alternate assignment, Moderators from api, and FAQ

### PR DESCRIPTION
**Pull request by endrawes0**

### CODE REVIEW

File: src/chesster.js [hunk 1]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+        return;
+    }
+    if (!channel) {
+        return;
+    }
+    var alternateOptions = message.league.options.alternate;
+    if (!alternateOptions || !_.isEqual(channel.name, alternateOptions.channel)) {
+        return;
+    }
+    var heltourOptions = message.league.options.heltour;
```
Comment (lakinwecker) [inline src/chesster.js]: We should refactor this into a middleware

File: src/chesster.js [hunk 2]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+
+    // Ensure the basic command format is valid
+    if (!_.isEqual(["assign", "to", "board", "during", "round", "on"], [assign, to, board, during, round, on])) {
+        replyMisunderstoodAlternateAssignment(bot, message);
+        return;
+    }
+
+    boardNumber = parseInt(boardNumber, 10);
+    if(isNaN(boardNumber)){
+        replyMisunderstoodAlternateUnassignement(bot, message);
```
Comment (lakinwecker) [inline src/chesster.js]: replyMisunderstoodAlternateAssignment or replyMisunderstoodAlternateUnassignment ??

File: src/chesster.js [hunk 3]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+/* look up the original player and assign him to his board */
+chesster.hears({
+    middleware: [ slack.withLeagueByChannelName ],
+    patterns: [ '^unassign' ],
+    messageTypes: [ 
+        'ambient'
+    ]
+}, 
+function(bot, message){
+    var channel = channels.byId[message.channel];
```
Comment (lakinwecker) [inline src/chesster.js]: This whole preamble feels like it should be a middleware.  But that's not required for this PR.

File: src/chesster.js [hunk 4]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+
+    boardNumber = parseInt(boardNumber, 10);
+    if(isNaN(boardNumber)){
+        replyMisunderstoodAlternateUnassignement(bot, message);
+        return;
+    }
+
+    roundNumber = parseInt(roundNumber, 10);
+    if(isNaN(roundNumber)){
+        replyMisunderstoodAlternatesUnassignment(bot, message);
```
Comment (lakinwecker) [inline src/chesster.js]: Same here.

File: src/league.js [hunk 5]
```diff
@@ -178,6 +190,16 @@ league_attributes = {
             self._teamLookup = newLookup;
         });
     },
+    
+    //--------------------------------------------------------------------------
+    // Figures out the current scheduling information for the round.
```
Comment (endrawes0) [inline src/league.js]: meant to change this comment

File: src/chesster.js [hunk 6]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+
+    // Ensure the basic command format is valid
+    if (!_.isEqual(["assign", "to", "board", "during", "round", "on"], [assign, to, board, during, round, on])) {
+        replyMisunderstoodAlternateAssignment(bot, message);
+        return;
+    }
+
+    boardNumber = parseInt(boardNumber, 10);
+    if(isNaN(boardNumber)){
+        replyMisunderstoodAlternateAssignement(bot, message);
```
Comment (lakinwecker) [inline src/chesster.js]: This has an extra 'e' in the Assign*e*ment

File: src/chesster.js [hunk 7]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+        return;
+    }
+
+    roundNumber = parseInt(roundNumber, 10);
+    if(isNaN(roundNumber)){
+        replyMisunderstoodAlternatesAssignment(bot, message);
```
Comment (lakinwecker) [inline src/chesster.js]: This one has an extra 's' in Alternate*s*Assignment

File: src/chesster.js (hunk 2 above)
Comment (lakinwecker) [inline src/chesster.js]: This one has an extra 'e' in replyMisunderstoodAlternateUnassign*e*ment

File: src/chesster.js (hunk 4 above)
Comment (lakinwecker) [inline src/chesster.js]: This one has an extra 's' in replyMisunderstoodAlternate*s*Unassignment

File: src/chesster.js [hunk 8]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+        during      = args[5], 
+        round       = args[6],
+        roundNumber = args[7],
+        on          = args[8];
+
+    var speaker = slack.getSlackUserFromNameOrID(message.user);
+    var speakerTeam = message.league.getTeamByPlayerName(speaker.name);
+
+    if (!isCaptainOrModerator(speaker, speakerTeam, teamName)) {
+        replyOnlyACaptainOrModeratorCanDoThat(bot, message);
```
Comment (lakinwecker) [inline src/chesster.js]: This one is missing an 'A' here: replyOnlyACaptainOr**A**ModeratorCanDoThat

File: src/chesster.js [hunk 9]
```diff
@@ -118,6 +118,197 @@ function leagueDMResponse(patterns, responseName) {
...
+        during      = args[5], 
+        round       = args[6],
+        roundNumber = args[7],
+        on          = args[8];
+
+    var speaker = slack.getSlackUserFromNameOrID(message.user);
+    var speakerTeam = message.league.getTeamByPlayerName(speaker.name);
+    
+    if (!isCaptainOrModerator(speaker, speakerTeam, teamName)) {
+        replyOnlyACaptainOrModeratorCanDoThat(bot, message);
```
Comment (lakinwecker) [inline src/chesster.js]: This one is missing an 'A' in replyOnlyACaptainOr**A**ModeratorCanDoThat

File: src/chesster.js [hunk 10]
```diff
@@ -408,14 +602,18 @@ function(bot, message) {
...
 function(bot, message) {
     bot_exception_handler(bot, message, function(){
-        if(_.isEqual(message.channel, channels.getId("general"))){
+        if(_.isEqual(message.channel, channels.getId(chesster.config["welcome"]["channel"]))){
             bot.reply(message, "Everyone, please welcome the newest member of the " 
                              + "Lichess 45+45 League, <@" + message.user + ">!");
             
             bot.startPrivateConversation(message, function(err, convo){
-               convo.say("Welcome. I'm the Lichess4545 League's Moderator Bot.");
+               convo.say("Welcome. I am the moderator bot for the Lichess4545 league");
```
Comment (lakinwecker) [inline src/chesster.js]: much better. :+1:

File: src/heltour.js [hunk 11]
```diff
@@ -166,7 +166,33 @@ function getPrivateURL(heltourConfig, page, user){
...
+    var request = heltourRequest(heltourConfig, "get_league_moderators");
+    request.parameters = {
+        "league": heltourConfig.leagueTag
+    };
+    return fetchJSONandHandleErrors(request).then(function(json){
+        return json["moderators"];
+    });
+}
 
+function fetchJSONandHandleErrors(request){
```
Comment (lakinwecker) [inline src/heltour.js]: Nice, I like this. We can (eventually) add in handling for non-200 errors and not trying to parse json if there is no body. Like the 405 error.

File: src/league.js [hunk 12]
```diff
@@ -164,7 +176,7 @@ league_attributes = {
             _.each(roster.teams, function(team) {
                 _.each(team.players, function(teamPlayer) {
                     var player = self.getPlayer(teamPlayer.username);
-                    player.isCaptain = player.is_captain;
+                    player.isCaptain = teamPlayer.is_captain;
```
Comment (lakinwecker) [inline src/league.js]: Yes! There is the bug. I didn't look hard, but I noticed this bug when I first wrote the code. Nice catch!

File: src/league.js [hunk 13]
```diff
@@ -576,7 +606,7 @@ league_attributes = {
     //--------------------------------------------------------------------------
     'formatModsResponse': function() {
         var self = this;
-        moderators = _.map(self.options.moderators, function(name) {
+        moderators = _.map(self._moderators, function(name) {
             return name[0] + "\u200B" + name.slice(1);
         });
         return Q.fcall(function() {
```
Comment (lakinwecker) [inline src/league.js]: I was thinking that this one should point out that it does _not_ ping the moderators, and should suggest the command that does ping them.  

```Lonewolf mods: endrawes0, jptriton, cyanfish, lakinwecker (**Note**: this does not ping the moderators. Use: `@chesster: summon mods` to ping them```

Comment (endrawes0) [inline src/league.js]: good idea

**Comment by endrawes0** [comment]
my god - what is with all the typos

**Comment by endrawes0** [comment]
Changes made. Merging.

**Pull request merged**